### PR TITLE
Add `trio-util` package

### PR DESCRIPTION
**Pull request by dhirschfeld**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml"
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example)
- [ ] Source is from official source
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged)
- [ ] If static libraries are linked in, the license of the static library is packaged.
- [x] Build number is 0
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details)
- [ ] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there

### CODE REVIEW

File: recipes/trio-util/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+    - python >=3.7
+    - pip
+  run:
+    - python >=3.7
+    - async_generator
+    - trio >=0.11.0
+
+test:
+  imports:
+    - trio_util
```
Comment (marcelotrevisani) [inline recipes/trio-util/meta.yaml:28]: can we try to add ``pip check`` here please?
```suggestion
    - trio_util
  requires:
    - pip
  command:
    - pip check
```

Comment (dhirschfeld) [reply recipes/trio-util/meta.yaml:28]: Sure - I think that's a good idea. Any reason it's not part of the example template?

**Comment by synapticarbors** [comment]
I opened up a PR upstream to include the license file in the source distribution in future releases: 
https://github.com/groove-x/trio-util/pull/11

**Comment by dhirschfeld** [comment]
Thanks @synapticarbors - that was on my TODO list!

**Comment by dhirschfeld** [comment]
NB: let me know if you'd like to be added as a maintainer!

**Comment by synapticarbors** [comment]
@dhirschfeld -- I'm a co-maintainer on ~120 feedstocks so I'll leave this one in your capable hands :-)

**Comment by dhirschfeld** [comment]
> I'm a co-maintainer on ~120 feedstocks so I'll leave this one in your capable hands 

Fair enough! 😅

**Comment by dhirschfeld** [comment]
It looks like the feedstock creation didn't work :(
* https://github.com/conda-forge/staged-recipes/runs/1312320503
```
+++set +x
+++python .travis_scripts/trigger_travis_build.py conda-forge/staged-recipes
b'access denied'
Traceback (most recent call last):
  File ".travis_scripts/trigger_travis_build.py", line 54, in <module>
    sys.exit(main(sys.argv))
  File ".travis_scripts/trigger_travis_build.py", line 47, in main
    rebuild_travis(args.slug)
  File ".travis_scripts/trigger_travis_build.py", line 38, in rebuild_travis
    response.raise_for_status()
  File "/home/travis/miniconda/lib/python3.7/site-packages/requests/models.py", line 941, in raise_for_status
    raise HTTPError(http_error_msg, response=self)
requests.exceptions.HTTPError: 403 Client Error: Forbidden for url: https://api.travis-ci.com/repo/conda-forge%2Fstaged-recipes/requests
```

image

**Comment by synapticarbors** [comment]
I just restarted it, so we'll see if it goes through this time.

**Comment by dhirschfeld** [comment]
That seems like an odd error:/

@synapticarbors - are you able to restart the CI?

**Comment by dhirschfeld** [comment]
Looks like you've done so - cheers!

**Comment by dhirschfeld** [comment]
For ref - the feedstock url:
https://github.com/conda-forge/trio-util-feedstock

**Comment by synapticarbors** [comment]
It looks like it's made it further this time

**Pull request merged**